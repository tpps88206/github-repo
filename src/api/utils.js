export const getHeaders = () => ({
  Accept: 'application/vnd.github.v3+json',
  'Content-Type': 'Application/json',
});
