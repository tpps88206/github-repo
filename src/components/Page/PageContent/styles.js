const styles = theme => ({
  root: {
    height: '100%',
  },
});

export default styles;
