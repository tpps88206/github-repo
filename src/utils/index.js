import isEmpty from 'lodash/isEmpty';
import split from 'lodash/split';
import upperCase from 'lodash/upperCase';

export const generateAvatarCharacter = fullName => {
  if (!fullName) {
    return '';
  }
  const repoName = split(fullName, '/', 2);
  if (repoName.length > 1) {
    return upperCase(repoName[1].substring(0, 1));
  }
  return upperCase(repoName[0].substring(0, 1));
};

export const generateQueryParams = params => {
  let result = '';
  if (isEmpty(params)) {
    return result;
  }

  params.forEach((param, index) => {
    if (index > 0) {
      result = result + '&';
    }
    result = result + param.key + '=' + param.value;
  });
  return result;
};
