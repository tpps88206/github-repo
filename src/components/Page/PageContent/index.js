import React from 'react';

import { makeStyles } from '@material-ui/core/styles';

import styles from './styles';

const useStyles = makeStyles(styles);

const PageContent = ({ children, className = '' }) => {
  const classes = useStyles();
  return <section className={`${classes.root} ${className}`}>{children}</section>;
};

export default PageContent;
