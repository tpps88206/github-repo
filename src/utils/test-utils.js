import React from 'react';
import { Provider } from 'react-redux';
// if you use Jest for testing and get the error with "use <Link> outside a <Router>"
// because just wrap your component with <BrowserRouter>
// see https://stackoverflow.com/questions/48640280/you-should-not-use-link-outside-a-router
import { BrowserRouter } from 'react-router-dom';

import { create } from 'jss';
import preset from 'jss-preset-default';

import { library } from '@fortawesome/fontawesome-svg-core';
import { fab } from '@fortawesome/free-brands-svg-icons';
import CssBaseline from '@material-ui/core/CssBaseline';
import { createMuiTheme, StylesProvider, ThemeProvider } from '@material-ui/core/styles';
import { render as rtlRender } from '@testing-library/react';

import theme from '@/constants/theme';
import configureStore from '@/redux/configureStore';
import styles from '@/styles';

library.add(fab);

const render = (ui, { ...renderOptions } = {}) => {
  const muiTheme = createMuiTheme(theme);
  const jss = create({
    plugins: [...preset().plugins],
  });
  const store = configureStore();

  jss.createStyleSheet(styles(muiTheme), { meta: 'global' }).attach();

  const Wrapper = ({ children }) => {
    return (
      <StylesProvider jss={jss}>
        <ThemeProvider theme={muiTheme}>
          <CssBaseline />
          <Provider store={store}>
            <BrowserRouter>{children}</BrowserRouter>
          </Provider>
        </ThemeProvider>
      </StylesProvider>
    );
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
