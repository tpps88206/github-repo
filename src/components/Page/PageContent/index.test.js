import React from 'react';

import { render } from '@/utils/test-utils';
import PageContent from './index';

describe('test Page Content', () => {
  it('renders PageContent without crashing', () => {
    const div = document.createElement('div');
    render(<PageContent />, div);
  });

  it('renders PageContent with snapshot', () => {
    const { asFragment } = render(<PageContent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
