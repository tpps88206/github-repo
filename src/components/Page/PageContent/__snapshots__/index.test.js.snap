// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Page Content renders PageContent with snapshot 1`] = `
<DocumentFragment>
  <section
    class="makeStyles-root-2 "
  />
</DocumentFragment>
`;
